src/shared/bootspec.c

```diff
@@ -1392,13 +1392,141 @@ static void boot_entry_file_list(
+        int r;
+
+        assert(e);
+
+        if (!strv_isempty(e->options)) {
+                _cleanup_free_ char *t = NULL, *t2 = NULL;
+                _cleanup_strv_free_ char **ts = NULL;
+
+                t = strv_join(e->options, " ");
+                if (!t)
+                        return log_oom();
+
+                ts = strv_split_newlines(t);
+                if (!ts)
+                        return log_oom();
+
+                t2 = strv_join(ts, "\n              ");
+                if (!t2)
+                        return log_oom();
+
+                printf("  defCmdline: %s\n", t2);
+                r = copy_cmdline(t2, &final_cmdline);
+                if (r < 0)
+                        return r;
+        }
+
+        LIST_FOREACH(addon_list, addon, e->local_addon_list) {
+                /* Add space at the beginning of addon_str to align it correctly */
+                print_addon(addon, " localAddon");
+                r = copy_cmdline(addon->cmdline, &final_cmdline);
+                if (r < 0)
+                        return r;
+        }
+
+        if (final_cmdline)
+                printf(" finalCmdline: %s\n", final_cmdline);
+
+        return 0;
+}
+
+static int json_addon(
+                BootEntryAddon *addon,
+                const char *addon_str,
+                JsonVariant **array) {
+
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        int r;
+
+        r = json_variant_merge_objectb(
+                        &v, JSON_BUILD_OBJECT(
+                                JSON_BUILD_PAIR(addon_str, JSON_BUILD_STRING(addon->location)),
+                                JSON_BUILD_PAIR("cmdline", JSON_BUILD_STRING(addon->cmdline))));
+        if (r < 0)
+                return log_oom();
+
+        r = json_variant_append_array(array, v);
+        if (r < 0)
+                return log_oom();
+
+        return 0;
+}
+
+static int json_cmdline(const BootEntry *e, JsonVariant **v) {
+
+        _cleanup_free_ char *final_cmdline = NULL;
+        _cleanup_free_ char *def_cmdline = NULL;
+        _cleanup_(json_variant_unrefp) JsonVariant *addons_array = NULL;
+        int r;
+
+        assert(e);
+
+        if (!strv_isempty(e->options)) {
+                def_cmdline = strv_join(e->options, " ");
+                if (!def_cmdline)
+                        return log_oom();
+                final_cmdline = strnappend(def_cmdline, " ", 1);
+        }
+
+        LIST_FOREACH(addon_list, addon, e->local_addon_list) {
+                r = json_addon(addon, "localAddon", &addons_array);
+                if (r < 0)
+                        return r;
+                r = copy_cmdline(addon->cmdline, &final_cmdline);
+                if (r < 0)
+                        return r;
+        }
+
+        r = json_variant_merge_objectb(
+                v, JSON_BUILD_OBJECT(
+                                JSON_BUILD_PAIR_CONDITION(def_cmdline, "defCmdline", JSON_BUILD_STRING(def_cmdline)),
```

Comment:
why is "defCmdline" supposed to mean?

please, no needless abbreviations. Lets not make things unnecnessarily cryptic

Reply:
maybe we should simply get rid of it and use finalCmdline to show the UKI .cmdline section? Not sure

Reply:
or ukiCmdline maybe